### PR TITLE
Adding remoteIP() and remotePort() functions

### PR DESCRIPTION
**Pull request by fredlcore**
These functions help to provide functionality such as IP-based access control to a server running on the Arduino for example.
Also, I have commented out two variables that create compiler warnings because they are not used.

### CODE REVIEW

File: src/WiFiSpiClient.h [hunk 1]
```diff
@@ -65,6 +65,12 @@ class WiFiSpiClient : public Client {
 
     using Print::write;
 
+    // Return the IP address of the host who sent the current incoming packet
+    virtual IPAddress remoteIP();
```
Comment (JiriBilek) [inline src/WiFiSpiClient.h]: Hi, I am not sure the functions should be _virtual_. Did you declare them so on purpose?

File: src/utility/espspi_proxy.h [hunk 2]
```diff
@@ -371,7 +371,7 @@ class EspSpiProxy
      */
     uint8_t crc8(uint8_t *buffer, uint8_t bufLen)
     {
-        const static uint8_t POLY = 0x07;
+//        const static uint8_t POLY = 0x07;
```
Comment (JiriBilek) [inline src/utility/espspi_proxy.h]: Thanks, it is a Copy-Paste issue. But next time, please, open another PR for changes not related to the original PR.

File: src/WiFiSpiClient.h (hunk 1 above)
Comment (fredlcore) [reply src/WiFiSpiClient.h]: Honestly, I don't know, I just copied them from the WiFiSpiUdp library and they work, but I'm not well versed with writing libraries, so "virtual ł could be unnecessary or even wrong.

File: src/utility/espspi_proxy.h (hunk 2 above)
Comment (fredlcore) [reply src/utility/espspi_proxy.h]: Sorry, don't know why it didn't end up in a separate pull request, sorry..

File: src/WiFiSpiClient.h (hunk 1 above)
Comment (JiriBilek) [reply src/WiFiSpiClient.h]: Ok, please remove the virtual keyword and commit it. I am not a C++ expert but I think there is no reason for this. Thanks

File: src/utility/espspi_proxy.h (hunk 2 above)
Comment (JiriBilek) [reply src/utility/espspi_proxy.h]: Please, delete the commented lines completely, there is no reason for them to remain here. Thanks.

Comment (fredlcore) [reply src/utility/espspi_proxy.h]: Done.

Comment (JiriBilek) [reply src/utility/espspi_proxy.h]: I don't see it, did you commit the change?

Comment (fredlcore) [reply src/utility/espspi_proxy.h]: Oops, didn't see the "delete the commented lines" comment, so this is not reflected yet, will do so now.

Comment (fredlcore) [reply src/utility/espspi_proxy.h]: Done, sorry for the typo, should read "unused variables" not "unused libraries".

**Review by JiriBilek — APPROVED** [review]
Now it's fine. Thanks

**Pull request merged**